core/src/main/scala/com/gu/AppIdentity.scala

```diff
@@ -67,14 +62,21 @@ object AppIdentity {
       autoScalingGroup.tags.asScala.map { t => t.key -> t.value }.toMap
     }
 
-    Option(EC2MetadataUtils.getInstanceId).map { instanceId =>
-      val tags = withOneOffAsgClient(client => getTags(client, instanceId))
-      AwsIdentity(
-        app = tags("App"),
-        stack = tags("Stack"),
-        stage = tags("Stage"),
-        region = EC2MetadataUtils.getEC2InstanceRegion
-      )
+    // Historically `getInstanceId` would return null if metadata service is unavailable (hence the Option), but
+    // this has changed and now throws an exception instead. So for safety we still handle all possibilities.
+    Try(Option(EC2MetadataUtils.getInstanceId)) match {
+      case Success(Some(instanceId)) =>
+        val tags = withOneOffAsgClient(client => getTags(client, instanceId))
+        Some(AwsIdentity(
+          app = tags("App"),
+          stack = tags("Stack"),
+          stage = tags("Stage"),
+          region = EC2MetadataUtils.getEC2InstanceRegion
+        ))
+      case Success(None) => None
+      case Failure(NonFatal(err)) =>
```

Comment:
I think Try will only catch NonFatal in the first place, it's only if you're using the built in try-catch you need that

Reply:
👍 good point